kafka-handler/src/test/org/apache/hadoop/hive/kafka/KafkaStorageHandlerInfoTest.java

```diff
@@ -0,0 +1,134 @@
+            final byte[] value = ("VALUE-" + Integer.toString(number)).getBytes(Charset.forName("UTF-8"));
+            return new ConsumerRecord<>(topic, 0, (long) number, 0L, null, 0L, 0, 0, KEY_BYTES, value);
+        }).collect(Collectors.toList());
+    }
+    private final Configuration conf = new Configuration();
+    private KafkaConsumer<byte[], byte[]> consumer;
+
+    private Properties consumerProps;
+
+    private static final String RESULT = "Partition(topic = TEST1, partition = 0, leader = 0, replicas = [0], isr = [0], offlineReplicas = [])" +
+        " [start offset = [0], end offset = [" + RECORD_NUMBER + "]]";
+
+    @BeforeClass
+    public static void setupCluster() throws Throwable {
+        KAFKA_BROKER_RESOURCE.before();
+        sendData();
+    }
+
+    @AfterClass
+    public static void tearDownCluster() {
+        KAFKA_BROKER_RESOURCE.deleteTopic(TOPIC);
+        KAFKA_BROKER_RESOURCE.after();
+    }
+
+    @Before
+    public void setUp() {
+        LOG.info("setting up Config");
+        setupConsumer();
+    }
+
+    @After
+    public void tearDown() {
+        consumer.close();
+        LOG.info("tearDown");
+    }
+
+    private void setupConsumer() {
+        this.conf.set(KafkaTableProperties.HIVE_KAFKA_BOOTSTRAP_SERVERS.getName(), KafkaBrokerResource.BROKER_IP_PORT);
+        consumerProps = new Properties();
+        consumerProps.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+        consumerProps.setProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "none");
+        consumerProps.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, KafkaBrokerResource.BROKER_IP_PORT);
+        consumerProps.setProperty(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class.getName());
+        consumerProps.setProperty(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class.getName());
+        //The configuration values are not default and are given randomly , these can be changed and tested if required
+        consumerProps.setProperty(ConsumerConfig.REQUEST_TIMEOUT_MS_CONFIG, "3002");
+        consumerProps.setProperty(ConsumerConfig.FETCH_MAX_WAIT_MS_CONFIG, "3001");
+        consumerProps.setProperty(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, "3001");
+        consumerProps.setProperty(ConsumerConfig.METADATA_MAX_AGE_CONFIG, "100");
+        this.consumer = new KafkaConsumer<>(consumerProps);
+    }
+
+    @Test
+    public void testFormatAsText() {
+        KafkaStorageHandlerInfo storageHandlerInfo = new KafkaStorageHandlerInfo(TOPIC, consumerProps);
+        Assert.assertEquals(String.class, storageHandlerInfo.formatAsText().getClass());
+        String text = storageHandlerInfo.formatAsText();
+        Assert.assertEquals(RESULT, text);
+    }
+
+    private static void sendData() {
+        List<ConsumerRecord<byte[], byte[]>> RECORDS = getRecords(TOPIC);
+        LOG.info("Setting up kafka producer");
```

Comment:
what about moving this log message after creating producerProps, and adding it too?

Reply:
done